cmd/juju/waitfor/unit.go

```diff
@@ -230,6 +230,11 @@ func (m UnitScope) GetIdents() []string {
 func (m UnitScope) GetIdentValue(name string) (query.Box, error) {
 	m.ctx.RecordIdent(name)
 
+	// UnitInfo could be nil - handle it here to avoid nil pointer dereference
+	if m.UnitInfo == nil {
+		return nil, query.RuntimeErrorf("UnitInfo is missing")
```

Comment:
I forced this error to happen and the output is weird:
```
ERROR Cannot execute query: runtime error.

1 | agent-status=="active"

wait-for requires the UnitInfo is missing
Maybe try removing the character and try again.
```

Reply:
bizarre, there's custom printing logic that checks the error type and prints that message.

I'll just use a fmt.Errorf then ?

Reply:
Changed it to this, I think it looks better
```
$ juju wait-for unit controller/0
ERROR internal error: UnitInfo is missing
```